KPI-team1-frontend/src/components/Sidebar.tsx

```diff
@@ -85,14 +95,25 @@ export default function SideBar({
               </NavLink>
 
               <div className="w-56 bg-[#F0F0F6] flex flex-col py-6 px-4 items-center gap-4 rounded-lg my-10">
-                <SearchBar isSearchKpi={isSearchKpi} />
+                <SearchBar
+                  isSearchKpi={isSearchKpi}
+                  setCircleId={setCircleId}
+                />
                 {circles &&
                   circles.map((circle, index) => (
                     <NavLink
+                      onClick={() =>
+                        setCircleId(circle.circle_user[0].circle_id)
+                      }
                       to={`/kpi/circles/${circle.circle_user[0].circle_id}`}
```

Comment:
I think it would be great if we could remain on the dashboard page when we switch circles and are already on the dashboard page - I think `useLocation` could come in handy here as well.